### DISCUSSION TITLE
Regarding Reading order instructions

### DISCUSSION THREAD

**Discussion by harsha20032020** [PaddlePaddle/PaddleOCR#13457] [Q&A]
I want to know if there is something similar to PSM modes in Tesseract in paddle OCR, Right now, the output is jumbled in my use case (which is newspapers , I've attached an example as well) 
![01012024-md-hr-15ll_segment_1_article](https://github.com/user-attachments/assets/4acc7e6b-3df3-4f95-b7de-b15713356549)
Where here the text between columns are jumbled, is there some way / options where i can collect the text outputted by the ocr model in a the order of reading.

Also might be a different question, but Is there any way I can speed up the inference process? Like Batching and on GPU? I've tried reading the docs but they are unclear on these details.

Thank you!!

**Comment by UserWangZz** [comment]
The definition of output in PaddleOCR is from top to bottom and from left to right. There are two solutions to the newspaper layout as shown in the figure:
1. Train a layout model, restore the layout based on the output of the layout model, and then use OCR for recognition;
2. Override the current sorting method and restore the OCR output to reading order.

**Comment by harsha20032020** [comment]
how to do the 2nd thing? Overriding current sorting method and restoring OCR output to reading order?

**Reply by UserWangZz** [reply]
Please reference this code in  predict_system.py
https://github.com/PaddlePaddle/PaddleOCR/blob/cf26f2330ec898a341dcdd2a37cdec56f7a1bf76/tools/infer/predict_system.py#L160-L182
